Sources/Commands/SwiftPackageTool.swift

```diff
@@ -183,7 +183,24 @@ public struct SwiftPackageTool: SwiftTool {
                     return try get(manifest, manifestParser: parseManifest)
                 }
             }
-        
+
+            func findDirectoryReferences(path: String) throws -> [String] {
+                let rootDirectories = walk(path, recursively: false)
+                let rootDirectoriesToConsider = rootDirectories.filter {
+                    if $0.hasSuffix(".xcodeproj") { return false }
+                    if $0.hasSuffix(".playground") { return false }
+                    if $0.basename.hasPrefix(".") { return false }
+                    return $0.isDirectory
+                }
+
+                return rootDirectoriesToConsider.filter {
+                    let directoriesWithSwift = walk($0).filter {
+                        $0.hasSuffix(".swift")
```

Comment:
True, it feels weird to have this stuff here, but my reasoning was that all the params used in `Xcodeproj.generate` are generated in [this file](https://github.com/apple/swift-package-manager/blob/master/Sources/Commands/SwiftPackageTool.swift#L237-L245) (talking about fetch, for instance), but now I realised that those are reused methods in other argument options.

Thanks for pointing me to `Sources.swift`!